gql_schema_codegen/dependency/dependency.py

```diff
@@ -1,12 +1,83 @@
 from itertools import groupby
-from typing import List, NamedTuple, Set
+from typing import Dict, List, NamedTuple, Set
 
 
 class Dependency(NamedTuple):
     imported_from: str
     dependency: str
 
 
+INTERMEDIATE_INTERFACES: Dict[str, str] = {}
+
+
+def get_interface_dependencies():
+    return INTERMEDIATE_INTERFACES
+
+
+def update_interface_dependencies(config_file_content):
+    global INTERMEDIATE_INTERFACES
+    if type(config_file_content) is dict:
```

Comment:
```suggestion
if isinstance(config_file_content, dict)
```

doing type check with `is` is not ideal:

```python
>>> class Foo(dict): pass
>>> type(Foo()) is dict
False
>>> isinstance(Foo(), dict)
True
```

Reply:
I know but I am explicitly followed the style of the rest of the repo here in the thought that we will perhaps refactor everything together

Reply:
if you do the `defaultdict` suggestion `isinstance` will be required. either way its up to you

Reply:
I think it makes sense to make the change - leaving extra bad code in was not great and leaves more work for potential future refactorings